mods/lord/Player/clans/init.lua

```diff
@@ -130,34 +152,39 @@ local clan_is_online_cache = {}
 --- @param name string
 --- @return boolean|nil
 function clans.clan_is_online(name)
-	if not clans.get_by_name(name) then return nil end
+	local clan = clans.get_by_name(name)
+	if not clan then return nil end
+	if clan.is_blocked then return nil end -- clan is offline if it is blocked
+
 	if clan_is_online_cache[name] == nil then
 		clan_is_online_cache[name] = check_clan_is_online(name)
 	end
 
 	return clan_is_online_cache[name]
 end
 
-minetest.register_on_joinplayer(function(player, last_login)
+minetest.register_on_joinplayer(function(player, _)
 	if not player or not player:is_player() then return end
 
+	--- @type Player
+	player = player -- HACK: makes VSCode type hints work
```

Comment:
хм... в ide-helper'е стоит указание, что это `Player`...
```lua
--- ...
--- ...
---
--- @param callback fun(player:ObjectRef|Player, last_login)
function minetest.register_on_joinplayer(callback) end
```
странно...
а если прям перед вызовом этой ф-ции поставить указание типа для `player` ?

```suggestion
--- @param player Player
minetest.register_on_joinplayer(function(player, _)
	if not player or not player:is_player() then return end

	--- @type Player
	player = player -- HACK: makes VSCode type hints work
```